include/z64.h

```diff
@@ -852,27 +849,51 @@ typedef struct {
 } PolygonDlist2; // size = 0x8
 
 typedef struct {
-    /* 0x00 */ u8    type;
+    /* 0x00 */ PolygonBase base;
     /* 0x01 */ u8    num; // number of dlist entries
     /* 0x04 */ void* start;
     /* 0x08 */ void* end;
 } PolygonType2; // size = 0xC
 
 typedef union {
-    Polygon      polygon;
+    PolygonBase  base;
     PolygonType0 polygon0;
     PolygonType1 polygon1;
     PolygonType2 polygon2;
-} Mesh; // "Ground Shape"
+} MeshHeader; // "Ground Shape"
+
+typedef enum {
+    /* 0 */ LENS_ACTORS_MODE_SEETHROUGH, // lens actors are visible by default and hidden by using lens (for example, fake walls)
+    /* 1 */ LENS_ACTORS_MODE_REVEAL // lens actors are invisible by default and revealed by using lens (for example, invisible enemies)
+} LensActorsMode;
+
+typedef enum {
+    /* 0 */ ROOM_BEHAVIOR_TYPE1_0,
+    /* 1 */ ROOM_BEHAVIOR_TYPE1_1,
+    /* 2 */ ROOM_BEHAVIOR_TYPE1_2,
+    /* 3 */ ROOM_BEHAVIOR_TYPE1_3, // unused
+    /* 4 */ ROOM_BEHAVIOR_TYPE1_4, // unused
+    /* 5 */ ROOM_BEHAVIOR_TYPE1_5
+} RoomBehaviorType1;
+
+typedef enum {
+    /* 0 */ ROOM_BEHAVIOR_TYPE2_0,
+    /* 1 */ ROOM_BEHAVIOR_TYPE2_1,
+    /* 2 */ ROOM_BEHAVIOR_TYPE2_2,
+    /* 3 */ ROOM_BEHAVIOR_TYPE2_3,
+    /* 4 */ ROOM_BEHAVIOR_TYPE2_4,
+    /* 5 */ ROOM_BEHAVIOR_TYPE2_5,
+    /* 6 */ ROOM_BEHAVIOR_TYPE2_6
+} RoomBehaviorType2;
 
 typedef struct {
     /* 0x00 */ s8   num;
     /* 0x01 */ u8   unk_01;
-    /* 0x02 */ u8   unk_02;
-    /* 0x03 */ u8   unk_03;
+    /* 0x02 */ u8   behaviorType2;
+    /* 0x03 */ u8   behaviorType1;
```

Comment:
im assuming the ordering in the function and the command itself 
```c
void Scene_CommandRoomBehavior(GlobalContext* globalCtx, SceneCmd* cmd) {
    globalCtx->roomCtx.curRoom.behaviorType1 = cmd->roomBehavior.gpFlag1;
    globalCtx->roomCtx.curRoom.behaviorType2 = cmd->roomBehavior.gpFlag2 & 0xFF;
```
ideally these will have more specific names in the future anyway, if its possible to do so depending on what they do

Reply:
I had another look, might be from the order in the `roomBehavior` struct:
```c
typedef struct {
    /* 0x00 */ u8  code;
    /* 0x01 */ u8  gpFlag1;
    /* 0x04 */ u32 gpFlag2;
} SCmdRoomBehavior;
```
I hope there is a clear separation between 1 and the bottom end of 2, although it's clear from the other uses in that function that a couple of the higher bits are used in an obvious way (and only a couple).

Reply:
At the end of the day it's arbitrary since as you guys noted they're undocumented and in inconsistent order in structs, but my motivation was mainly `behaviorType1` seems more important than `behaviorType2`

In particular `behaviorType1` is responsible for a few barely noticeable changes, like in dungeons (`ROOM_BEHAVIOR_TYPE1_1` I think) the ztarget sound or sun song not reloading the scene.

Besides `func_808409CC` in z_player and "hot room" checks, `behaviorType2` has no use, but maybe that z_player code is *really* important idk.

I do think we can find better names once the values are documented. Which imo is made easier by using those temporary not-so-bad-if-they-were-to-stay names